alibi/explainers/backends/pytorch/cfrl_base.py

```diff
@@ -510,46 +512,98 @@ def update_actor_critic(ae: AE,
     return losses
 
 
-def to_numpy(x: Optional[Union[List, np.ndarray, torch.Tensor]]) -> Optional[Union[List, np.ndarray]]:
+def to_numpy(X: Optional[Union[List, np.ndarray, torch.Tensor]]) -> Optional[Union[List, np.ndarray]]:
     """
     Converts given tensor to numpy array.
 
     Parameters
     ----------
-    x
+    X
         Input tensor to be converted to numpy array.
 
     Returns
     -------
     Numpy representation of the input tensor.
     """
-    if x is not None:
-        if isinstance(x, np.ndarray):
-            return x
+    if X is not None:
+        if isinstance(X, np.ndarray):
+            return X
 
-        if isinstance(x, torch.Tensor):
-            return x.detach().cpu().numpy()
+        if isinstance(X, torch.Tensor):
+            return X.detach().cpu().numpy()
 
-        if isinstance(x, list):
-            return [to_numpy(e) for e in x]
+        if isinstance(X, list):
+            return [to_numpy(e) for e in X]
 
-        return np.array(x)
+        return np.array(X)
 
     return None
 
 
-def to_tensor(x: Union[np.ndarray, torch.Tensor], device: torch.device, **kwargs) -> Optional[torch.Tensor]:
+def to_tensor(X: Union[np.ndarray, torch.Tensor], device: torch.device, **kwargs) -> Optional[torch.Tensor]:
     """
     Converts tensor to torch.Tensor
 
     Returns
     -------
     torch.Tensor conversion.
     """
-    if x is not None:
-        if isinstance(x, torch.Tensor):
-            return x.to(device)
+    if X is not None:
+        if isinstance(X, torch.Tensor):
+            return X.to(device)
 
-        return torch.tensor(x).to(device)
+        return torch.tensor(X).to(device)
 
     return None
+
+
+def save_model(path: Union[str, os.PathLike], model: nn.Module) -> None:
+    """
+    Saves a model and its optimizer.
+
+    Parameters
+    ----------
+    path
+        Path to the saving location.
+    model
+        Model to be saved.
+    """
+    torch.save(model, path)
+
+
+def load_model(path: Union[str, os.PathLike]) -> nn.Module:
+    """
+    Loads a model and its optimizer.
+
+    Parameters
+    ----------
+    path
+        Path to the loading location.
+
+    Returns
+    -------
+    Loaded model.
+    """
+    model = torch.load(path)
+    model.eval()
+    return model
+
+
+def seed(seed: int = 13):
+    """
+    Sets a seed to ensure reproducibility
+    Parameters
+    ----------
+    seed
+        seed to be set
+    """
+
+    # torch related
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
+
+    # others
+    np.random.seed(seed)
+    random.seed(seed)
```

Comment:
These three functions feel like they belong into a generic `pytorch` utility module so can be reused for other stuff in the future?

Reply:
Yes. The functions are generic.